src/app/components/Application/Application.jsx

```diff
@@ -38,13 +38,19 @@ class App extends React.Component {
 
   render() {
     let heroData;
-    const type = !_isEmpty(this.props.params) ? this.props.params.type : undefined;
+    let heroDOM;
+    // Use pathname index as type since routes.jsx specifies them directly.
+    let type = this.props.location.pathname.split('/')[4];
```

Comment:
Probably to have a check here to see if `this.props.location.pathname` is a string.